infrastructure/smart-contracts/contracts/upgrades/referee/Referee9.sol

```diff
@@ -424,16 +425,61 @@ contract Referee9 is Initializable, AccessControlEnumerableUpgradeable {
         bytes32 comboHash = keccak256(abi.encodePacked(_assertionId, rollupAddress));
         require(!rollupAssertionTracker[comboHash], "9");
         rollupAssertionTracker[comboHash] = true;
+        
+        // Connect to the rollup contract
+        IRollupCore rollup = IRollupCore(rollupAddress);
+
+        // get the node information from the rollup.
+        Node memory currentNode = rollup.getNode(_assertionId);
+
+
+
+        // If the gap is more than 1 assertion, we need to handle as a batch challenge
+        if(_assertionId - _predecessorAssertionId > 1){
+
+            // Initialize the array to store the assertionIds
+            uint64 [] memory assertionIds = new uint64[](_assertionId - _predecessorAssertionId);
+
+            // Loop through the assertions and check if they have been submitted
+            for(uint64 i = _predecessorAssertionId + 1; i < _assertionId; i++){
+
+                // create the comboHash for the assertionId and rollupAddress
+                bytes32 comboHashBatch = keccak256(abi.encodePacked(i, rollupAddress));
+
+                // check the assertionId and rollupAddress combo haven't been submitted yet
+                require(!rollupAssertionTracker[comboHashBatch], "15");
+
+                // set the comboHash to true to indicate it has been submitted
+                rollupAssertionTracker[comboHashBatch] = true;
+
+                // add the assertionId to the array
+                assertionIds[i - _predecessorAssertionId - 1] = i;
+            }
+            
+            // Verify Batch ConfirmData
+            (, bytes32 confirmHash) = RefereeCalculations(refereeCalculationsAddress).getConfirmDataMultipleAssertions(assertionIds, rollupAddress);
+            require(_confirmData == confirmHash, "11");
+       
+            // emit the batch challenge event
+            emit BatchChallenge(challengeCounter, assertionIds);
+        }
 
         // verify the data inside the hash matched the data pulled from the rollup contract
         if (isCheckingAssertions) {
 
+            // Last challenge assertionId submitted
+            uint64 lastChallengeAssertionId = challenges[challengeCounter - 1].assertionId;
+
             // get the node information from the rollup.
-            Node memory node = IRollupCore(rollupAddress).getNode(_assertionId);
+            Node memory previousNode = rollup.getNode(lastChallengeAssertionId);
+
+            require(previousNode.prevNum == _predecessorAssertionId, "10");
+            require(currentNode.createdAtBlock == _assertionTimestamp, "12");
 
-            require(node.prevNum == _predecessorAssertionId, "10");
-            require(node.confirmData == _confirmData, "11");
-            require(node.createdAtBlock == _assertionTimestamp, "12");
+            // Verify Individual ConfirmData
+            if(_assertionId - _predecessorAssertionId == 1){
+                require(currentNode.confirmData == _confirmData, "11");
+            }            
```

Comment:
It looks like we want to do something like this in the loop for each assertion id we submit for, make sure the `_predecessorAssertionId` is the assertion id from the last challenge and that we now submit for the current assertionId with the correct data, then if its more than one, the confirmData has to match with all the assertion confirmHash on the rollup

Reply:
@CryptITAustria lets jump on a call to discuss this one. 

_predecessorAssertionId is already being checked here on  478

`require(previousNode.prevNum == _predecessorAssertionId, "10");`

Batched confirm data is being checked here on 467:

`require(_confirmData == confirmHash, "11");`